cli/flox-rust-sdk/src/providers/services.rs

```diff
@@ -274,31 +269,53 @@ impl ProcessStates {
             .map(|state| state.name.clone())
             .collect()
     }
+}
 
-    fn fmt_table(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        writeln!(f, "{:<20} {:<10} {:>8}", "NAME", "STATUS", "PID")?;
-        for proc in &self.0 {
-            writeln!(f, "{:<20} {:<10} {:>8}", proc.name, proc.status, proc.pid)?;
-        }
-        Ok(())
+#[derive(Clone, Debug, Serialize)]
+/// Simplified version of ProcessState for display in the CLI.
+pub struct ProcessStateDisplay {
+    name: String,
+    status: String,
+    pid: u64,
+}
+
+#[derive(Clone, Debug, Serialize)]
+/// Simplified version of ProcessStates for display in the CLI.
+pub struct ProcessStatesDisplay(Vec<ProcessStateDisplay>);
+
+impl From<ProcessStates> for ProcessStatesDisplay {
+    fn from(procs: ProcessStates) -> Self {
+        ProcessStatesDisplay(
+            procs
+                .0
+                .into_iter()
+                .sorted_by_key(|proc| proc.name.clone())
+                .map(|proc| ProcessStateDisplay {
+                    name: proc.name,
+                    status: proc.status,
+                    pid: proc.pid,
+                })
+                .collect(),
+        )
     }
+}
 
-    fn fmt_json_lines(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        for proc in &self.0 {
-            let line = serde_json::to_string(proc).map_err(|_| std::fmt::Error)?;
-            writeln!(f, "{line}")?;
-        }
-        Ok(())
+impl IntoIterator for ProcessStatesDisplay {
+    type IntoIter = std::vec::IntoIter<ProcessStateDisplay>;
+    type Item = ProcessStateDisplay;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.into_iter()
     }
 }
 
-impl Display for ProcessStates {
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        if f.alternate() {
-            self.fmt_json_lines(f)
-        } else {
-            self.fmt_table(f)
+impl Display for ProcessStatesDisplay {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        writeln!(f, "{:<20} {:<10} {:>8}", "NAME", "STATUS", "PID")?;
+        for proc in &self.0 {
+            writeln!(f, "{:<20} {:<10} {:>8}", proc.name, proc.status, proc.pid)?;
         }
```

Comment:
suggestion: pad to the longest name

```suggestion
        let longest = self.0.iter().map(|state| state.name.len()).max().unwrap_or(20);
        writeln!(f, "{:<longest$} {:<10} {:>8}", "NAME", "STATUS", "PID")?;
        for proc in &self.0 {
            writeln!(f, "{:<longest$} {:<10} {:>8}", proc.name, proc.status, proc.pid)?;
        }
```

Reply:
Pushed a variation of this because we need to account for the name being shorter than the header.